Playback Effects Voiceover crash fix

bjtitus opened the pull request:
Fixes a crash when using Voiceover to change the Trim Silence amount beyond its bounds.

Also fixes an issue where this segmented control wasn't added to the Voiceover list after enabling the Trim Silence switch.

## To test

* Open the fullscreen player and open Playback Effects
* Enable Voice Over using Siri
* Swipe through to ensure all controls are selectable
* Enable "Trim Silence"
* ✅ Ensure the Trim Silence values are selectable
* Try changing the values 
* ✅ Ensure values can be changed by swiping without crashing the app

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

bjtitus committed:
Fix crash with accessibility value changes in CustomSegmentedControl

bjtitus committed:
Add Trim Silence segmented control when enabled

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/105">7.101 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

danielebogo reviewed (commented):
@bjtitus the app is not crashing but I'm not able to change the trim silence value once I select medium.

bjtitus committed:
Fix selection of Trim Silence with VoiceOver

This moves selections state handling to separate actionViews in the segmented control

danielebogo reviewed (changes requested):
@bjtitus I can now navigate trough the trim silence options. However I think the PR needs to be rebased on the release version and not trunk, unless you want to change target
